coordination_network_toolkit/preprocess.py

```diff
@@ -107,62 +109,64 @@ def preprocess_twitter_json_data(db_path: str, tweets: Iterable[str]):
-        if "full_text" in tweet:
-            tweet_text = tweet["full_text"]
-        elif "extended_tweet" in tweet:
-            tweet_text = tweet["extended_tweet"]["full_text"]
-        else:
-            tweet_text = tweet["text"]
-
-        retweet = tweet.get("retweeted_status", {})
-
-        row = (
-            tweet["id_str"],
-            tweet["user"]["id_str"],
-            tweet["user"]["screen_name"],
-            retweet.get("id_str"),
-            tweet.get("in_reply_to_status_id_str", None),
-            tweet_text,
-            len(tweet_text),
-            zlib.adler32(tweet_text.encode("utf8")),
-            # This will be populated only when similarity calculations are necessary
-            None,
-            # Twitter epoch in seconds
-            int((int(tweet["id"]) >> 22) / 1000),
-        )
+    try:
+        db.execute("begin")
 
-        db.execute(
-            "insert or ignore into edge values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)", row,
-        )
+        for raw_tweet in tweets:
 
-        # If it's a retweet, don't consider any of the urls as candidates
-        if not row[3]:
-            if retweet:
-                url_entities = retweet.get("extended_tweet", retweet)["entities"][
-                    "urls"
-                ]
-            else:
-                url_entities = tweet.get("extended_tweet", tweet)["entities"]["urls"]
+            tweet = json.loads(raw_tweet)
 
-            message_id, user_id = row[:2]
-            timestamp = row[-1]
-
-            urls = [u["expanded_url"] for u in url_entities]
-            # Ignore urls shared in reposts
-            for url in urls:
-                db.execute(
-                    "insert or ignore into message_url values(?, ?, ?, ?)",
-                    (message_id, url, timestamp, user_id),
-                )
-
-    db.execute("commit")
-    db.close()
\ No newline at end of file
+            # Try grabbing the full_text field from the extended format, otherwise
+            # check if there's a extended_tweet object.
+            # print(sorted(tweet.keys()))
+            if "full_text" in tweet:
+                tweet_text = tweet["full_text"]
+            elif "extended_tweet" in tweet:
+                tweet_text = tweet["extended_tweet"]["full_text"]
+            else:
+                tweet_text = tweet["text"]
+
+            retweet = tweet.get("retweeted_status", {})
+
+            row = (
+                tweet["id_str"],
+                tweet["user"]["id_str"],
+                tweet["user"]["screen_name"],
+                retweet.get("id_str"),
+                tweet.get("in_reply_to_status_id_str", None),
+                tweet_text,
+                len(tweet_text),
+                zlib.adler32(tweet_text.encode("utf8")),
+                # This will be populated only when similarity calculations are necessary
+                None,
+                # Twitter epoch in seconds
+                int((int(tweet["id"]) >> 22) / 1000),
```

Comment:
For consistency with the other preprocessing it might make sense to just take the inner float.

Reply:
as in just `(int(tweet["id"]) >> 22) / 1000` without the outer int()?

Reply:
Made the change.

Reply:
Yep that's what I meant :)